--- src/Terraria/Terraria/Graphics/Renderers/LegacyPlayerRenderer.cs
+++ src/tModLoader/Terraria/Graphics/Renderers/LegacyPlayerRenderer.cs
@@ -5,6 +_,7 @@
 using Terraria.DataStructures;
 using Terraria.GameContent;
 using Terraria.ID;
+using Terraria.ModLoader;
 
 namespace Terraria.Graphics.Renderers
 {
@@ -35,6 +_,8 @@
 			_gore.Clear();
 			PlayerDrawHeadSet drawinfo = default(PlayerDrawHeadSet);
 			drawinfo.BoringSetup(drawPlayer, _drawData, _dust, _gore, position.X, position.Y, alpha, scale);
+
+			/*
 			PlayerDrawHeadLayers.DrawPlayer_00_BackHelmet(ref drawinfo);
 			PlayerDrawHeadLayers.DrawPlayer_01_FaceSkin(ref drawinfo);
 			PlayerDrawHeadLayers.DrawPlayer_02_DrawArmorWithFullHair(ref drawinfo);
@@ -44,6 +_,14 @@
 			PlayerDrawHeadLayers.DrawPlayer_06_NormalHats(ref drawinfo);
 			PlayerDrawHeadLayers.DrawPlayer_07_JustHair(ref drawinfo);
 			PlayerDrawHeadLayers.DrawPlayer_08_FaceAcc(ref drawinfo);
+			*/
+
+			var layers = PlayerHooks.GetDrawHeadLayers(drawPlayer);
+
+			foreach (var layer in layers) {
+				layer.Draw(ref drawinfo);
+			}
+
 			CreateOutlines(alpha, scale, borderColor);
 			PlayerDrawHeadLayers.DrawPlayer_RenderAllLayers(ref drawinfo);
 		}
