--- src/Terraria/Terraria/DataStructures/PlayerDrawSet.cs
+++ src/tModLoader/Terraria/DataStructures/PlayerDrawSet.cs
@@ -8,16 +_,17 @@
 
 namespace Terraria.DataStructures
 {
+	//This had to be turned into a class to allow us to rename fields with the help of ref properties. Not ideal. --Mirsario.
-	public struct PlayerDrawSet
+	public class PlayerDrawSet
 	{
 		public List<DrawData> DrawDataCache;
 		public List<int> DustCache;
 		public List<int> GoreCache;
 		public Player drawPlayer;
 		public float shadow;
-		public Vector2 Position;
+		public Vector2 position;
 		public int projectileDrawPosition;
-		public Vector2 ItemLocation;
+		public Vector2 itemLocation;
 		public int armorAdjust;
 		public bool missingHand;
 		public bool missingArm;
@@ -76,9 +_,9 @@
 		public Color legsGlowColor;
 		public Color ArkhalisColor;
 		public float stealth;
-		public Vector2 legVect;
+		public Vector2 legOrigin;
-		public Vector2 bodyVect;
+		public Vector2 bodyOrigin;
-		public Vector2 headVect;
+		public Vector2 headOrigin;
 		public Color selectionGlowColor;
 		public float torsoOffset;
 		public bool hidesTopSkin;
@@ -115,6 +_,14 @@
 		public bool drawUnicornHorn;
 		public Color floatingTubeColor;
 		public Vector2 helmetOffset;
+
+#pragma warning disable IDE1006 // Naming Styles
+		internal ref Vector2 Position => ref position;
+		internal ref Vector2 ItemLocation => ref itemLocation;
+		internal ref Vector2 legVect => ref legOrigin;
+		internal ref Vector2 bodyVect => ref bodyOrigin;
+		internal ref Vector2 headVect => ref headOrigin;
+#pragma warning restore IDE1006 // Naming Styles
 
 		public Vector2 Center => new Vector2(Position.X + (float)(drawPlayer.width / 2), Position.Y + (float)(drawPlayer.height / 2));
 
